### PR TITLE
Add a solidus compare script

### PR DESCRIPTION
**Pull request by blocknotes**
It will allow us to monitor the changes on a Solidus project path.

## Usage
- To list the changes in diff format: `bin/solidus_compare`
- Apply the required changes
- To update the ignores list (ignore key in _config/solidus_compare.yml_): `bin/solidus_compare -u`

## Description
To keep our code in sync with the latest changes on Solidus frontend we can use this tool to list what changed.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bin/solidus_compare [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  puts "\n> #{path} changed files"
+  puts list
+  path
+end.compact
+
+if diff.any?
+  path = diff.first
+  cmd = "git diff master:#{path} remotes/#{REMOTE_SOURCE}/master:frontend/#{path}"
+  puts "\n#{cmd}\n  to check the changes"
+end
```
Comment (kennyadsl) [inline bin/solidus_compare:36]: What about pushing the output to a temp file, that we can also use to push the output of the diff in the [CircleCI artifacts tab](https://circleci.com/docs/2.0/artifacts/#artifacts-overview)?

Comment (blocknotes) [reply bin/solidus_compare:36]: Nice idea 👍 
I'm still looking for an easy to read format for differences. But we can start with just a diff a see how it goes

**Comment by kennyadsl** [comment]
This is great, but I don't understand how can we know something changed. Shouldn't the check be red when new changes are detected?

**Comment by blocknotes** [comment]
> This is great, but I don't understand how can we know something changed. Shouldn't the check be red when new changes are detected?

I preferred to avoid blocking the CI workflow, but I'm not completely convinced about this choice currently 😄

**Pull request merged**